sys-boot/refind/refind-0.11.4.ebuild

```diff
@@ -0,0 +1,197 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs flag-o-matic multiprocessing
+
+DESCRIPTION="The UEFI Boot Manager by Rod Smith"
+HOMEPAGE="https://www.rodsbooks.com/refind/"
+
+SRC_URI="mirror://sourceforge/project/${PN}/${PV}/${PN}-src-${PV}.tar.gz"
+
+LICENSE="BSD GPL-2 GPL-3 FDL-1.3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+FS_USE="btrfs +ext2 +ext4 hfs +iso9660 ntfs reiserfs"
```

Comment:
What is exactly the purpose of that whole fs hackery? Does it have any real gain, or does it merely make the result more 'pretty' at the cost of adding lot of useless complexity to the ebuild?

Reply:
Each file system support can be built or not, so the need to have there name in the use flags. As there is a loop in `src_prepare` to decide which file system should be built, it is easier to separate the fs use flags from other ones. I can then use `FS_USE` in `src_prepare`.

Reply:
But is there a value in not building all of them unconditionally?

Reply:
Well, here are the values I can think about:
- time and space saving (even if it is not that big);
- it is expected by upstream;
- there used to be failing fs building in previous versions, so users may want to disable compilation of these fs if not needed for them;
- gentoo users like to control everything they build on their system… ;)

Reply:
If upstream expects it, then why do you need to patch it in?

Reply:
If we want to build only selected file systems using upstream `Makefile`, we have to directly call the `Makefile` which is in the `filesystem` directory, either providing the variable containing the wanted fs, or calling `make` for each fs one after the other. But this requires to select the way the compilation is made (either `gnuefi` or `udk`) by ourselves, because this selection is made only in the main `Makefile`.

For a matter of consistency, I prefer to let the main `Makefile` always choose the way everything is compiled, which then requires to update the fs variable in the `Makefile` of `filesystem`.

One thing I could do anyway is not removing the fs from `Refind.dsc` (which always build all). In this case, for `udk` style building, it would build every fs, but install only selected ones. But the cost of removing the fs is low, and it increase _safetyness_ in case the compilation does not work for a particular unneeded fs.

Reply:
How do the two 'ways' of compiling differ? Is the end result different?

Compilation not working is not an argument. USE flags are not workarounds for sloppy packages. You are supposed to test them all, and make sure all of them work.

Reply:
Yes, the end result is certainly different, because the compilation is not based on the same libraries.

Regarding compilation not working, OK, I understand. So I should not patch `Refind.dsc`.

Reply:
If the result is different, then `||` is not acceptable and you must either settle on one, or provide explicit choice between the two (i.e. USE flags).

As for filesystems, I believe you should remove the flags and just build them unconditionally unless you can either make it work without sed, or find value that really justifies that amount of extra code.

Reply:
OK, I will keep using the `gnuefi` use flag, then. And use Makefile variables only for filesystems.

Reply:
Done.